docs/versioning_vendor_api.md

```diff
@@ -0,0 +1,138 @@
+# Versioning Vendor API
+
+## Summary
+
+This document describes the steps required in order to increment the minor version of the Vendor API
+
+
+## Introducing new changes
+
+Changes in the API are specified in `VersionChange` classes and each version can be made up of one or more classes. Every change class must be configured in `app/lib/vendor_api.rb`
+
+A `VersionChange` class and has a `description`, an `action` and/or one or multiple resources set using the `resource` identifier.
+
+
+### Adding a new endpoint
+
+New endpoints are not by default made available via the API. In order to make them available from a specific version onwards, the endpoint's action must be configured in the version class and the version class must be mapped on a version.
+
+e.g. to make retrieving all applications available in version 1.1
+
+
+in `/app/lib/vendor_api/changes/retrieve_applications.rb`
+
+```ruby
+class RetrieveApplications < VersionChange
+   action ApplicationsController, :index
+end
+```
+
+in `/app/lib/vendor_api,rb`
+
+```ruby
+module VendorAPI
+  ...
+  VERSIONS = {
+   '1.0' => [ ... ],
+   '1.1' => [ Changes::RetrieveApplications ]
+  }
+end
+```
+
+### Adding a new resource
+
+To ensure that the API supports all existing minor versions, any changes must be introduced through presenter modules. This enables serving older versions without including changes introduced at later points.
+
+#### Adding a new presenter
+
+Presenters must extend the `VendorAPI::Base` and return the object Hash through a private `schema` method. The hash can then be converted to JSON through public methods that utilise the application cache (look at `ApplicationPresenter`).
+
+Below, you can see an example of configuring a new `InterviewPresenter`
+
+_configure the new presenter in the VersionChange class_
+
+```ruby
+class AddInterviews < VersionChange
+   resource InterviewPresenter
+end
+```
+
+```ruby
+module VendorAPI
+  ...
+  VERSION = {
+   '1.0' => [...],
+   '1.1' => [AddInterviews],
+  }
+end
+```
+
+#### Updating an existing presenter
+
+When introducing new functionality or changing the behavior of existing fields, create a new module that implements the `schema` method, and deep merge into the returned value. Changing the values directly in the existing presenter is not recommended as that will have an impact on previous versions of the API.
```

Comment:
Would be really great to see an example of a `schema` here too! How does a `schema` relate to the OpenAPI schema?

Reply:
The `schema` method here isn't related to the OpenAPI schema, but to how we build and present the application JSON and to allow merging in new fields or updating existing one.

We kept the documentation decoupled from the actual implementation, but will think of automation in the next iterations